tensorflow/docs_src/deploy/s3.md

```diff
@@ -1,12 +1,57 @@
 # How to run TensorFlow on S3
 
-This document describes how to run TensorFlow on S3 file system.
+Tensorflow supports reading and writing data to S3. S3 is an object storage API which is nearly ubiquitious, and has many available implementations:
 
-## S3
+* [Amazon S3](https://aws.amazon.com/s3/)
+* [Google Storage](https://cloud.google.com/storage/docs/interoperability)
+* [Minio](https://www.minio.io/kubernetes.html)
+* [Ceph](http://docs.ceph.com/docs/master/radosgw/)
 
-We assume that you are familiar with @{$reading_data$reading data}.
+This document guides you through the required setup, and provides examples on usage.
 
-To use S3 with TensorFlow, change the file paths you use to read and write
+## Setup
+
+To configure your Tensorflow applications to communicate to an S3 endpoint, you must set the following environment variables:
+
+```bash
+AWS_ACCESS_KEY_ID=XXXXX                 # Credentials only needed if connecting to a private endpoint
+AWS_SECRET_ACCESS_KEY=XXXXX
```

Comment:
I think `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` is not a must as you could allow credentials to be setup through other means. In fact, it might not be desired as it is visible in env.

Reply:
I've made a note of this below, but wanted people to be able to copy/paste to get something working quickly.

I can expand the description here to include a warning about secrets as env vars, and also mention the improved security of the other methods.

Reply:
Thanks @elsonrodriguez. I am thinking we keep the original descriptions of environmental variables and setting credential alone. Instead, we could create another section of example at the beginning. In the example section we could copy the setup you added (with ENV for `AWS_ACCESS_KEY_ID`/etc) and the example you added.

Reply:
I've added back the section, however it feels redundant, should I get rid of the variable comment descriptions in the setup section?

Reply:
Thanks @elsonrodriguez. As most of the ENVs are not necessary (e.g., `S3_USE_HTTPS` will be used unless `S3_USE_HTTPS=0` is provided), maybe we could remove most of the ENVs here.

We could keep the ENVs of `AWS_ACCESS_KEY_ID=XXXXX` and `AWS_SECRET_ACCESS_KEY=XXXXX` in the example. As the user will need to provide credentials one way (ENV) or another (profiles, etc).

Reply:
If you put it that way, I think maybe it's fine as is.

Most of my testing lately has been with Minio, which I deploy without SSL/HTTPS. If I were quickly skimming this example I might miss that setting and be confused.

I think I'm ok with the redundancy given the clarity in both the example and configuration guide.

Reply:
Thanks @elsonrodriguez. If the current documentation format helps I think we can leave it as it. 
Though the usage of word "must" might be strong. Can you change the sentence to:
`To configure your Tensorflow applications to communicate to an S3 endpoint, you could set the following environment variables:`

Also, is it possible to move the `Configuration Notes` section before the `Setup` section? I am concerned that asking users to provide `ENV` explicitly at the beginning of the doc may mislead and make users to believe that `ENVs` are necessary. It would be better served as only an "example setup" I think.

Reply:
That makes sense, I rearranged the doc and change the sentence to be softer and more contextual.